test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/JQueryQueryStringValueProviderFactoryTest.cs

```diff
@@ -17,6 +17,12 @@ public class JQueryQueryStringValueProviderFactoryTest
     {
         private static readonly Dictionary<string, StringValues> _backingStore = new Dictionary<string, StringValues>
         {
+            // Reviewers: A fair number of cases in the following dataset seem impossible for jQuery.ajax() to send
+            // given how $.ajax() and $.param() are defined i.e. $.param() won't add a bare name after ']'. Also, the
+            // names we generate for <input/> elements are always valid. Should we remove these weird / invalid cases
+            // e.g. normalizing "property[]Value"? (That normalizes to "propertyValue".) See also
+            // https://github.com/jquery/jquery/blob/1ea092a54b00aa4d902f4e22ada3854d195d4a18/src/serialize.js#L13-L92
+            // The alternative is to handle "[]name" and "[index]name" cases while normalizing keys.
```

Comment:
These comments won't go into `dev`. Thoughts on the alternatives? Of course, the other option is to leave "impossible" cases alone.

Reply:
I think we should leave them alone, or possibly move them to another test that has a comment that explains that these are tests for cases we don't think will occur in nature.

It's important if you're parsing something to process some invalid cases also :)